crypto/bn/bn_prime.c

```diff
@@ -458,16 +420,23 @@ static int probable_prime(BIGNUM *rnd, int bits, prime_t *mods)
     return 1;
 }
 
-int bn_probable_prime_dh(BIGNUM *rnd, int bits,
-                         const BIGNUM *add, const BIGNUM *rem, BN_CTX *ctx)
+static int probable_prime_dh(BIGNUM *rnd, int bits, int safe, prime_t *mods,
+                             const BIGNUM *add, const BIGNUM *rem,
+                             BN_CTX *ctx)
 {
     int i, ret = 0;
     BIGNUM *t1;
+    BN_ULONG delta;
+    BN_ULONG maxdelta = BN_MASK2 - primes[NUMPRIMES - 1];
 
     BN_CTX_start(ctx);
     if ((t1 = BN_CTX_get(ctx)) == NULL)
         goto err;
 
+    if (maxdelta > BN_MASK2 - BN_get_word(add))
+        maxdelta = BN_MASK2 - BN_get_word(add);
```

Comment:
`add` might not fit in a word.

Reply:
In which case `maxdelta` should end up as 0.

Reply:
Yes, in that case BN_get_word(add) will return BN_MASK2, so maxdelta = 0.
I assume add will not be negative, though.

Reply:
Right, and the loop will only run once, and goto again